pkg/worker/worker.go

```diff
@@ -122,6 +125,18 @@ func (w *Worker) Daemon() {
 		go w.daemon(h)
 	}
 
+	// Signal the worker daemon's readiness by closing the internal ready channel.
+	// This mechanism implies that Worker.Daemon() must never be called twice,
+	// because closing a closed channel results in a runtime panic. Time based
+	// systems are often a source of race conditions. Providing this mechanism may
+	// help facilitate e.g. unit tests concerned with concurrency patterns, so
+	// that we do not have to rely on time based systems within event driven
+	// problem domains.
+
+	{
+		close(w.rdy)
+	}
```

Comment:
Adding this here since I noticed that the time based approach wasn't right in the tests. In general it is a good pattern to provide a listener for async work to have finished. We are not exposing it via interface right now, but we could if we had to.